x-pack/plugins/lens/public/shared_components/coloring/color_ranges/color_ranges_validation.tsx

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { i18n } from '@kbn/i18n';
+import { isValidColor } from '../utils';
+
+import type { ColorRange, ColorRangeAccessor } from './types';
+
+/** @internal **/
+type ColorRangeValidationErrors = 'invalidColor' | 'invalidValue' | 'greaterThanMaxValue';
+
+/** @internal **/
+export interface ColorRangeValidation {
+  errors: ColorRangeValidationErrors[];
+  isValid: boolean;
+}
+
+/** @internal **/
+export const getErrorMessages = (colorRangesValidity: Record<string, ColorRangeValidation>) => {
+  return [
+    ...new Set(
+      Object.values(colorRangesValidity)
+        .reduce<ColorRangeValidationErrors[]>((acc, item) => [...acc, ...item.errors], [])
+        .flat()
+    ),
+  ].map((item) => {
+    switch (item) {
+      case 'invalidColor':
+      case 'invalidValue':
+        return i18n.translate('xpack.lens.dynamicColoring.customPalette.invalidValueOrColor', {
+          defaultMessage: 'At least one color range contains the wrong value or color',
+        });
+      case 'greaterThanMaxValue':
+        return i18n.translate('xpack.lens.dynamicColoring.customPalette.invalidMaxValue', {
+          defaultMessage: 'Maximum value should be greater than preceding values',
+        });
+      default:
+        return '';
+    }
+  });
+};
+
+/** @internal **/
+export const validateColorRange = (colorRange: ColorRange, accessor: ColorRangeAccessor) => {
+  const errors: ColorRangeValidationErrors[] = [];
+
+  if (Number.isNaN(colorRange[accessor])) {
+    errors.push('invalidValue');
+  }
+
+  if (accessor === 'end') {
+    if (colorRange.start > colorRange.end) {
+      errors.push('greaterThanMaxValue');
+    }
+  } else {
+    if (!isValidColor(colorRange.color)) {
```

Comment:
Add a check here for the `invalidValue` already added.

Reply:
not sure that it's right, I've fixed `getErrorMessages` to avoid of duplicating messages